pymeasure/instruments/instrument.py

```diff
@@ -126,50 +141,70 @@ class Instrument:
 
     # Variable holding the list of DynamicProperty parameters that are configurable
     # by users
-    _fget_params_list = ('get_command',
-                         'values',
-                         'map_values',
-                         'get_process',
-                         'command_process',
-                         'check_get_errors')
-
-    _fset_params_list = ('set_command',
-                         'validator',
-                         'values',
-                         'map_values',
-                         'set_process',
-                         'command_process',
-                         'check_set_errors')
+    _fget_params_list = (
+        "get_command",
+        "values",
+        "map_values",
+        "get_process",
+        "command_process",
+    )
+
+    _fset_params_list = (
+        "set_command",
+        "validator",
+        "values",
+        "map_values",
+        "set_process",
+        "command_process",
+    )
 
     # Prefix used to store reserved variables
     __reserved_prefix = "___"
 
+    @property
+    @abc.abstractmethod
+    def id_starts_with(self):
```

Comment:
The `abstractproperty` decorator has been [deprecated ](https://docs.python.org/3/library/abc.html#abc.abstractproperty)- this will essentially have the same functionality